WcaOnRails/app/models/competition.rb

```diff
@@ -331,6 +331,22 @@ def warnings_for(user)
       unless has_any_round_per_event? && schedule_includes_rounds?
         warnings[:schedule] = I18n.t('competitions.messages.schedule_must_match_rounds')
       end
+
+      if championship_warnings.any?
+        warnings = championship_warnings.merge(warnings)
+      end
+    end
+
+    warnings
+  end
+
+  def championship_warnings
+    warnings = {}
+    for championship in self.championships
```

Comment:
Nit: `self.championships.each do |championship|` is what we consequently use instead of for loops ^^

Reply:
Done.